domains/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -216,84 +216,50 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        /// Updates the light client state
         #[pallet::weight(T::WeightInfo::sync_committee_period_update())]
         #[transactional]
-        pub fn sync_committee_period_update(
+        #[pallet::call_index(0)]
+        pub fn light_client_update(
             origin: OriginFor<T>,
-            sync_committee_period_update: SyncCommitteePeriodUpdateOf<T>,
+            light_client_update: LightClientUpdateOf,
         ) -> DispatchResult {
             let _sender = ensure_signed(origin)?;
 
             Self::check_bridge_blocked_state()?;
 
-            let sync_committee_period = sync_committee_period_update.sync_committee_period;
+            let sync_committee_period = light_client_update.sync_committee_period;
             log::info!(
                 target: "ethereum-beacon-client",
-                "💫 Received sync committee update for period {}. Applying update",
+                "💫 Received light client update for sync committee period {}. Applying update",
                 sync_committee_period
             );
 
-            if let Err(err) =
-                Self::process_sync_committee_period_update(sync_committee_period_update)
-            {
+            Self::process_light_client_update(light_client_update).map_err(|err| {
                 log::error!(
                     target: "ethereum-beacon-client",
-                    "💫 Sync committee period update failed with error {:?}",
+                    "💫 Light client update failed with error {:?}",
                     err
                 );
-                return Err(err);
-            }
+                err
+            })?;
 
             log::info!(
                 target: "ethereum-beacon-client",
-                "💫 Sync committee period update for period {} succeeded.",
+                "💫 Light client update for sync committee period {} succeeded.",
                 sync_committee_period
             );
 
             Ok(())
         }
 
-        #[pallet::weight(T::WeightInfo::import_finalized_header())]
-        #[transactional]
-        pub fn import_finalized_header(
```

Comment:
Do we not need this anymore ?

Reply:
This is not supposed to be a separate endpoint as per the ethereum specification. The `light client update` endpoint updates the entire state of light client (except execution header import).